tests/components/admin_console/manage_teams_modal/manage_teams_modal.test.jsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import ManageTeamsModal from 'components/admin_console/manage_teams_modal/manage_teams_modal.jsx';
+
+describe('ManageTeamsModal', () => {
+    function emptyFunc() {} // eslint-disable-line no-empty-function
+
+    const baseProps = {
+        onModalDismissed: emptyFunc,
+        show: true,
+        user: {
+            id: 'currentUserId',
+            last_picture_update: '1234',
+            email: 'currentUser@test.com',
+            roles: 'system_user',
+            username: 'currentUsername',
+        },
+        updateTeamMemberSchemeRoles: emptyFunc,
+        getTeamMembersForUser: emptyFunc,
+        getTeamsForUser: emptyFunc,
+    };
+
+    test('should match snapshot init', () => {
+        const getTeamMembersForUser = jest.fn();
+        const getTeamsForUser = jest.fn();
+
+        const wrapper = shallow(
+            <ManageTeamsModal
+                {...baseProps}
+                getTeamMembersForUser={getTeamMembersForUser}
+                getTeamsForUser={getTeamsForUser}
+            />
+        );
+
+        expect(wrapper).toMatchSnapshot();
+        expect(getTeamMembersForUser).toHaveBeenCalledTimes(1);
+        expect(getTeamMembersForUser).toHaveBeenCalledWith('currentUserId');
+        expect(getTeamsForUser).toHaveBeenCalledTimes(1);
+        expect(getTeamsForUser).toHaveBeenCalledWith('currentUserId');
+    });
+
+    test('should save data in state from api calls', async () => {
+        const mockTeamData = {
+            id: '123test',
+            name: 'testTeam',
+            display_name: 'testTeam',
+            delete_at: 0,
+        };
+
+        const getTeamMembersForUser = async () => {
+            return {
+                data: [{team_id: '123test'}],
+            };
+        };
+
+        const getTeamsForUser = async () => {
+            return {
+                data: [mockTeamData],
+            };
+        };
+
+        const wrapper = shallow(
+            <ManageTeamsModal
+                {...baseProps}
+                getTeamMembersForUser={getTeamMembersForUser}
+                getTeamsForUser={getTeamsForUser}
+            />
+        );
+
+        await getTeamsForUser();
+        await getTeamMembersForUser();
+
+        wrapper.update();
+        expect(wrapper.state('teams')).toEqual([mockTeamData]);
+        expect(wrapper.state('teamMembers')).toEqual([{team_id: '123test'}]);
+
+        expect(wrapper).toMatchSnapshot();
+    });
```

Comment:
Above works but we could do it as well like:
```
const getTeamMembersForUser = jest.fn().mockImplementation(() => {
    return Promise.resolve({data: [{team_id: '123test'}]});
});

const getTeamsForUser = jest.fn().mockImplementation(() => {
    return Promise.resolve({data: [mockTeamData]});
});

const wrapper = shallow(
    <ManageTeamsModal
        {...baseProps}
        getTeamMembersForUser={getTeamMembersForUser}
        getTeamsForUser={getTeamsForUser}
    />
);

process.nextTick(() => {
    expect(wrapper.state('teams')).toEqual([mockTeamData]);
    expect(wrapper.state('teamMembers')).toEqual([{team_id: '123test'}]);
    expect(wrapper).toMatchSnapshot();
});
```

Reply:
Thanks. That's neat!

I like the above implementation when i am trying to check for mocked values but here it was covered in first test case so, i am good for now with the way it is.

I am happy to consider as i am 0/5 about the changing the tests here unless you have any stronger opinion on it.

Reply:
I understand but in this case I prefer to change it as I posted. I feel like it's the way to uncover all the power of Jest.

Also, I like to see that the components are normally doing its lifecycle without manually calling the mock function and/or `wrapper.update`.

Reply:
```javascript

Reply:
Thanks @sudheerDev for taking time to consider my suggestion, and from our offline chat, there are some assertion error / unable to exit test suite when you tried as I posted posted above.
With that, I totally agree with your findings and test implementation.